c10/core/dispatch/OpSchema.h

```diff
@@ -32,16 +29,17 @@ inline DeviceTypeId to_device_type_id(DeviceType device_type) {
   }
 }
 
-// TODO get rid of tensor_to_dispatch_key once c2::Tensor is de-templatized. This then fits into a template lambda instead of a functor.
-struct tensor_to_dispatch_key final {
-    template<class TensorType>
-    TensorParameterDispatchKey operator()(const TensorType& tensor) const {
-      return TensorParameterDispatchKey{
-          to_device_type_id(tensor.GetDeviceType()),
-          LayoutId(0),
-          tensor.dtype().id()};
-    }
-};
+inline TensorParameterDispatchKey tensor_to_dispatch_key(const C10Tensor& tensor) {
+  return TensorParameterDispatchKey{
+      to_device_type_id(tensor.impl()->device_type()),
+      LayoutId(0),
```

Comment:
This doesn't look right.

Reply:
What's wrong with it? Are you commenting on the change or was it already wrong before? Note that this code is still part of the old dispatcher design, this diff is not about changing the dispatcher design, that'll come in future diffs.

Reply:
LayoutId(0) is hardcoded to 0 and when I see a hardcoded number like that, it looks wrong.